internal/security/setup/option/optionHelpers_test.go

```diff
@@ -0,0 +1,153 @@
+//
+// Copyright (c) 2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License. You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License
+// is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+// or implied. See the License for the specific language governing permissions and limitations under
+// the License.
+//
+// SPDX-License-Identifier: Apache-2.0'
+//
+
+package option
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/edgexfoundry/edgex-go/internal/security/setup"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestGetWorkDir(t *testing.T) {
+	// sets up configuration for SecretsSetupInfo
+	tearDown := setupGenerateTest(t)
+	defer tearDown()
+
+	origEnvVal := os.Getenv(envXdgRuntimeDir)
+	os.Unsetenv(envXdgRuntimeDir)
+
+	// test WorkDir that's configured by toml
+	runTimeDir := getWorkDir()
+	expectedWorkDir, _ := filepath.Abs("./workingtest")
+	assert.Equal(t, expectedWorkDir, runTimeDir)
+
+	// test default WorkDir
+	oldConfig := setup.Configuration
```

Comment:
I like having `defer tearDown()` like elsewhere in the codebase here rather than just re-assign later because when you modify the global value, the cleanup of that variable/state is right next to the mocking and you don't have to remember to go looking for the cleanup/re-assignment later down in the test.

Reply:
ditto for the other tests in this file

Reply:
not sure i understand the comment above. 
if you are saying that it's better to do the re-assignment of setup.Configuration in the teardown(), rather than in each of the unit tests, then i agree.
otherwise, could you clarify?

Reply:
Yes either in the existing teardown function or in a new teardown function